tests/src/JIT/Directed/Misc/Concat/ConcatTest.cs

```diff
@@ -0,0 +1,66 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+internal class ConcatTest
+{
+    static string strA = "A";
+    static string strB = "B";
+    static string strC = "C";
+    static string strD = "D";
+    static string strE = "E";
+
+    static string strAB      = "AB";
+    static string strABC     = "ABC";
+    static string strABCD    = "ABCD";
+    static string strABCDE   = "ABCDE";
+    static string strABCDx2  = "ABCDABCD";
+
+    static int iReturn = 100;
+
+    static public int Main()
+    {
+        iReturn = 100;
+        try
+        {
+            string result;
+
+            result = string.Concat(strA, strB);
```

Comment:
Since this is an automatic C# compiler thing, I don't think that the API test coverage in CoreCLR repo would create a test case with source code to call to Concat directly with 5 arguments since it doesn't see such an API when it reflects over the API space for String.